pkg/extgrpc/client.go

```diff
@@ -48,13 +54,14 @@ func StoreClientGRPCOpts(logger log.Logger, reg *prometheus.Registry, tracer ope
 		reg.MustRegister(grpcMets)
 	}
 
-	if !secure {
+	// Insecure if secure is false and no TLS config is supplied.
+	if !secure && (tlsConfig == store.TLSConfiguration{}) {
```

Comment:
one way of improving this would be maybe doing `secure  = tlsConfig == store.TLSConfiguration{}`